### DISCUSSION TITLE
Row level security error until first refresh

### DISCUSSION THREAD

**Discussion by awwsome** [supabase/supabase#7770] [Questions]
### Problem

Whenever I try to insert some data into the table, it fails due to RLS error.
```
code: "42501"
details: null
hint: null
message: "new row violates row-level security policy for table \"post\""
```
### Strange thing
After I refresh the browser, everything works fine.
Inserting data fails in following situations:

- Log out → Quit browser → Relaunch browser → Log in → Insert → ❌
- (Check inserting works on current tab) → Log out → Open a new tab or window →  Log in → Insert → ❌
- After successful inserting → Several minutes passes without doing anything → Insert → ❌

### What I did

1. Created a table, enabled RLS, and set up policies as follows:
<img width="491" alt="image" src="https://user-images.githubusercontent.com/6214507/179552496-42fa05ce-972c-441e-b118-67121d4ade3c.png">

2. Found an [article](https://nikofischer.com/supabase-error-row-level-security-policy) describing the similar issue, and I passed `returning: 'minimal'` as advised.

I guess this has to do with session or authentication thing, but I am too newbie to figure out what is problem.
Thank you.

**Comment by GaryAustin1** [comment]
You are likely running your operation before the client code has finished setting up the session, either from local storage, or having to refresh the token.   You should console.log your session/user right before your operation to see if it is there.
